integrationtests/testground/main.go

```diff
@@ -109,14 +109,20 @@ func testSyncThreads(env *runtime.RunEnv, ic *run.InitContext) (err error) {
 	setup(env, ic)
 	for i, pair := range simulatedNetworks {
 		round := pair.simulation
+		timeout, err := time.ParseDuration(env.StringParam("test-timeout"))
+		if err != nil {
+			timeout = time.Minute
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
 		if ic.GlobalSeq == 1 {
 			msg("################### Round %s network ###################", round)
-			ic.NetClient.MustConfigureNetwork(context.Background(), &network.Config{
-				Network:       "default",
-				Enable:        true,
-				Default:       pair.shape,
-				CallbackState: sync.State("network-configured-" + round),
-				RoutingPolicy: network.AllowAll,
+			ic.NetClient.MustConfigureNetwork(ctx, &network.Config{
+				Network:        "default",
+				Enable:         true,
+				Default:        pair.shape,
+				CallbackState:  sync.State("network-configured-" + round),
+				CallbackTarget: 1,
```

Comment:
This was what caused the test never succeeded on docker/k8s. By default it waits for `env.TestInstanceCount` replies, but we only call this in one instance.